### PR TITLE
Paraview Purge as part of Qt4 Removal

### LINKED ISSUE #30500: Remove paraview/VSI (SaveVTK should stay)
Remove paraview/VSI (SaveVTK should stay)

### PR DESCRIPTION
**Pull request by Pasarus**
**Description of work.**
I am removing Paraview, VSI and Surrounding supported code. This includes DynamicPDF Interface. This is a part of the removal of Qt4

**To test:**
Ensure Qt help docs open on MacOS and Windows
Ensure Packaging works and installs properly of MacOS and Windows

Ubuntu and RHEL7/CentOS are tested by Builds and myself.

Fixes #30500 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
@peterfpeterson @quantumsteve I think I remember you mentioning DynamicPDF at some point? Is this still required?

File: Framework/Kernel/src/UsageService.cpp [hunk 1]
```diff
@@ -273,9 +273,6 @@ std::string UsageServiceImpl::generateStartupMessage() {
   message["osVersion"] = ConfigService::Instance().getOSVersion();
   message["osReadable"] = ConfigService::Instance().getOSVersionReadable();
 
-  // legacy interface requires paraview version
-  message["ParaView"] = 0;
```
Comment (peterfpeterson) [inline Framework/Kernel/src/UsageService.cpp:277]: This cannot go. I don't believe the webapi knows what to do when this key is missing

File: Framework/Kernel/src/UsageService.cpp [hunk 2]
```diff
@@ -273,9 +273,6 @@ std::string UsageServiceImpl::generateStartupMessage() {
   message["osVersion"] = ConfigService::Instance().getOSVersion();
   message["osReadable"] = ConfigService::Instance().getOSVersionReadable();
 
-  // legacy interface requires paraview version
```
Comment (martyngigg) [inline Framework/Kernel/src/UsageService.cpp:276]: It's quite likely we'll need to keep this as the web api might require the payload to contain it. We should check first.

File: Framework/Kernel/src/ErrorReporter.cpp [hunk 3]
```diff
@@ -96,9 +96,6 @@ std::string ErrorReporter::generateErrorMessage() const {
   message["osVersion"] = ConfigService::Instance().getOSVersion();
   message["osReadable"] = ConfigService::Instance().getOSVersionReadable();
 
-  // legacy interface requires paraview version
```
Comment (martyngigg) [inline Framework/Kernel/src/ErrorReporter.cpp:99]: It's quite likely we'll need to keep this as the web api might require the payload to contain it. We should check first.

**Review by peterfpeterson — CHANGES_REQUESTED** [review]

**Comment by peterfpeterson** [comment]
Dynamic PDF gui is a thing that we are going to rewrite in qtpy. It's on the list, but not urgent to get done right now.

**Comment by Pasarus** [comment]
@peterfpeterson Is it ok to remove in this PR do you think? Considering it is not exposed to workbench at present, removing this alongside Paraview/Vates allows me to remove extra files that support only that and Paraview/Vates, it makes sense to me.

File: Framework/Kernel/src/UsageService.cpp (hunk 2 above)
Comment (Pasarus) [reply Framework/Kernel/src/UsageService.cpp:276]: Thank you for letting me know. I will add it back :+1:

File: Framework/Kernel/src/ErrorReporter.cpp (hunk 3 above)
Comment (Pasarus) [reply Framework/Kernel/src/ErrorReporter.cpp:99]: Thank you for letting me know. I will add it back :+1:

File: Framework/Kernel/src/UsageService.cpp (hunk 1 above)
Comment (Pasarus) [reply Framework/Kernel/src/UsageService.cpp:277]: Thank you for letting me know. I will add it back :+1:

**Review by DanielMurphy22 — DISMISSED** [review]
Packages built on Jenkins for Mac and Windows both work well! I assume the points by @martyngigg and @peterfpeterson have been reolved.

File: Framework/Kernel/test/ErrorReporterTest.h [hunk 4]
```diff
@@ -51,9 +51,9 @@ class ErrorReporterTest : public CxxTest::TestSuite {
     reader.parse(message, root);
     auto members = root.getMemberNames();
     const std::vector<std::string> expectedMembers{
-        "ParaView", "application", "host",       "mantidSha1", "mantidVersion",
```
Comment (martyngigg) [inline Framework/Kernel/test/ErrorReporterTest.h:54]: As we're leaving the ParaView key in I think we should still check for it in the test?

**Review by martyngigg — CHANGES_REQUESTED** [review]

File: Framework/Kernel/test/ErrorReporterTest.h (hunk 4 above)
Comment (Pasarus) [reply Framework/Kernel/test/ErrorReporterTest.h]: Should be back in now, great catch :+1:

File: Framework/Kernel/test/UsageServiceTest.h [hunk 5]
```diff
@@ -70,8 +70,9 @@ class UsageServiceTest : public CxxTest::TestSuite {
     reader.parse(message, root);
     auto members = root.getMemberNames();
     std::vector<std::string> expectedMembers{
-        "ParaView", "application", "host",       "mantidSha1", "mantidVersion",
-        "osArch",   "osName",      "osReadable", "osVersion",  "uid"};
+        "application",   "host",      "mantidSha1",
```
Comment (martyngigg) [inline Framework/Kernel/test/UsageServiceTest.h:73]: `ParaView` here too?

Comment (Pasarus) [reply Framework/Kernel/test/UsageServiceTest.h]: Another great catch! :+1:

**Review by martyngigg — APPROVED** [review]

**Comment by martyngigg** [comment]
I think this is certainly the bulk of it and anything else left is stragglers that we can deal with at a later time.

**Pull request merged**